test/common/network/connection_impl_test.cc

```diff
@@ -232,11 +238,11 @@ class ConnectionImplTest : public testing::TestWithParam<Address::IpVersion> {
           return new Buffer::WatermarkBuffer(below_low, above_high, above_overflow);
         }));
 
-    Event::MockTimer* timer = nullptr;
     if (create_timer) {
       // This timer will be returned (transferring ownership) to the ConnectionImpl when
       // createTimer() is called to allocate the delayed close timer.
-      timer = new Event::MockTimer(dispatcher.get());
+      timer_ = new Event::MockTimer(dispatcher.get());
+      timer_->timer_destroyed_ = &timer_destroyed_;
```

Comment:
Well, clang-tidy is not really happy with the way we manage memory here so this makes clang-tidy happy and I also think this is correct.

The way it works right now is:

1. `ConnectionImplTest::createConnectionMocks` allocates a MockTimer via `new`
2. `MockTimer` programs the dispatcher `createTimer_` to return the `MockTimer` and this call also changes the ownership of the `MockTimer` from `ConnectionImplTest` to the caller of `createTimer_`
3. Memory gets deleted

Clang tidy detects this as a memory leak and treats this as an error. I speculate that I am the first person that changed this file after we updated clang and this is why I stumbled on this error.

With this change, I store the pointer in the class and make sure that it got deallocated by the caller of `createTimer_`, otherwise, I clean it up myself.

Reply:
`timer_destroyed_` is a boolean. I think you want `timer_destroyed_ = true`?

Reply:
I feel like I am missing something and this is a slippery slope to making a fool of myself in some way :)

The destructor of the `MockTimer` is;

```
MockTimer::~MockTimer() {
  if (timer_destroyed_) {
    *timer_destroyed_ = true;
  }
}
```

So when this is destructor is called it sets: `ConnectionImplTest::timer_destroyed_ = true` and then we know that the timer has been destroyed and that `~ConnectionImplTest` doesn't have to do that. (a bit orthogonal but we should probably also add an assert to ensure that `timer_destroyed_ == true` because we know that we pass the ownership)

Reply:
OK sorry I didn't realize one is a pointer and one is a boolean. OK this makes sense. Does the case in `TEST_P(ConnectionImplTest, ServerTransportSocketTimeoutDisabledOnConnect)` also need to be updated?

Reply:
This test allocates their own Timer and they could also leak it if `MockTimer` ownership is never transferred. However, the test does assert that `timer_destroyed_ == true`. 

I fix it as well however, I don't think that the leak even matters that much. I am surprised though that clang-tidy doesnt have a problem with this test.